### PR TITLE
Add R.without

### PR DESCRIPTION
**Pull request by raine**
As I proposed on gitter,

```js
without :: a -> [a] -> [a]
```

### CODE REVIEW

**Comment by asaf-romano** [comment]
I'd personally prefer a version taking a list as the first argument, similar to ``omit``.

**Comment by davidchambers** [comment]
> I'd personally prefer a version taking a list as the first argument, similar to `omit`.

I had the same thought, then realized this would be [`R.difference`][1] with arguments flipped. @raine, is `without` significantly more convenient than `difference` in the situations in which you've used it?


[1]: http://ramdajs.com/docs/#difference

**Comment by raine** [comment]
`R.difference` returns a set, i.e. removes duplicate values

**Comment by davidchambers** [comment]
Good point!

**Comment by CrossEye** [comment]
:herb:

**Comment by asaf-romano** [comment]
Just this morning I used R.difference as a substitute for R.without (the list variant), and as always got confused with the arguments order. I guess that's because``R.difference`` is a binary function in which both arguments are considered "data" and have the same shape, albeit the order matters.  What I really want to do is to express this idea in a pipeline:

``R.without(["_id", "__version__"]) // filter db stuff``

**Comment by davidchambers** [comment]
I don't have a view as to whether the first argument should be of type `a` or `[a]`. If you believe `a` to be the better option, @raine, I'm happy for you to merge this pull request.

**Comment by raine** [comment]
I updated this to take a list as first argument. Please have another look.

File: src/without.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * @category List
+ * @sig [a] -> [a] -> [a]
+ * @param {Array} list1 The values to be removed from `list2`.
+ * @param {Array} list2 The array to remove values from.
+ * @return {Array} The new array without values in `list1`.
+ * @example
+ *
+ *      R.without([1, 2], [1, 2, 1, 3, 4]); //=> [3, 4]
+ */
+module.exports = useWith(reject, [flip(_contains), identity]);
```
Comment (davidchambers) [inline src/without.js]: I'd prefer not to use `useWith` internally. A lambda would reduce the number of Ramda functions involved in this definition, reducing the footprint of `require('ramda/src/without')` when using Browserify or similar.

File: src/without.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+var _contains = require('./internal/_contains');
+var flip = require('./flip');
+var identity = require('./identity');
+var reject = require('./reject');
+var useWith = require('./useWith');
+
+/**
```
Comment (davidchambers) [inline src/without.js]: Nit: Two empty lines after imports, please. :)

File: test/without.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    };
+
+    eq(R.without([0], [-0]).length, 1);
+    eq(R.without([-0], [0]).length, 1);
+    eq(R.without([NaN], [NaN]).length, 0);
+    eq(R.without([[1]], [[1]]).length, 0);
+    eq(R.without([new Just([42])], [new Just([42])]).length, 0);
+  });
+});
+
```
Comment (davidchambers) [inline test/without.js]: :space_invader:

File: src/without.js (hunk 1 above)
Comment (CrossEye) [inline src/without.js]: That's interesting.  While you're probably right, I originally created `useWith` in order to elegantly write `project`.

File: src/without.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ * Returns a new list without values in the first argument.
+ * `R.equals` is used to determine equality.
+ *
+ * Acts as a transducer if a transformer is given in list position.
+ * @see R.transduce
```
Comment (davidchambers) [inline src/without.js]: Please move this line to between the `@return` and `@example` annotations (see #1458).

**Comment by CrossEye** [comment]
This has been sitting a while.  Any objections?

**Comment by davidchambers** [comment]
:deciduous_tree:

**Pull request merged**